### PR TITLE
fix(web): KMW recorder modularization, restoration 🧩

### PR DESCRIPTION
**Pull request by jahorton**
As KMW's recorder module also includes the code to reproduce recorded test sequences, modularizing and restoring it is a prerequisite to restoring KMW's pre-existing integration-test suite.

Also includes a few bug-fixes that were interfering with the process, such as properly deferring keyboard-stub addition that occurs before the engine initializes.

@keymanapp-test-bot skip

### CODE REVIEW

File: web/src/app/browser/src/keymanEngine.ts [hunk 1]
```diff
@@ -241,6 +242,57 @@ export default class KeymanEngine extends KeymanEngineBase<BrowserConfiguration,
...
+    return Lr;
+  }
+
+  /**
+   * Get API-friendly array of available keyboard stubs
+   *
+   * @return   {Array}     Array of available keyboards
+   *
+   */
+  getKeyboards() {
```
Comment (mcdurdin) [inline web/src/app/browser/src/keymanEngine.ts:280]:
```suggestion
  public getKeyboards() {
```

Helping to indicate that this is a public API

File: web/src/app/browser/src/keymanEngine.ts [hunk 2]
```diff
@@ -241,6 +242,57 @@ export default class KeymanEngine extends KeymanEngineBase<BrowserConfiguration,
...
+    Lr['Font'] = Lstub['KFont'];
+    Lr['OskFont'] = Lstub['KOskFont'];
+    Lr['HasLoaded'] = !!Lkbd;
+
+    Lr['IsRTL'] = Lkbd ? !!Lkbd['KRTL'] : null;
+    return Lr;
+  }
+
+  /**
+   * Get API-friendly array of available keyboard stubs
```
Comment (mcdurdin) [inline web/src/app/browser/src/keymanEngine.ts:275]:
```suggestion
   * Get API-friendly array of available keyboard stubs
   *
   * API Version 17.0
```

Need something to indicate this is a published API function?

**Review by mcdurdin — APPROVED** [review]

File: web/src/app/browser/src/keymanEngine.ts (hunk 1 above)
Comment (jahorton) [reply web/src/app/browser/src/keymanEngine.ts]: I've added a tracking item on #8560 regarding this.  While I've made the change suggested here, I get the feeling that this and the other comment are intended to be generalized beyond the scope of this PR.

Comment (mcdurdin) [reply web/src/app/browser/src/keymanEngine.ts]: > I've added a tracking item on #8560 regarding this. While I've made the change suggested here, I get the feeling that this and the other comment are intended to be generalized beyond the scope of this PR.

Yes, that's a good feeling to have here :grin:

**Pull request merged**